### DISCUSSION TITLE
File comitted and ignored in submodule

### DISCUSSION THREAD

**Discussion by chrjabs** [release-plz/release-plz#1905] [Q&A] [answered]
Hey, in a project using `release-plz` that I'm working on, I have a repository included as a submodule that I don't have control over. (It's somebody else's repository that I'm using as a dependency.) In said submodule there are two files that are committed but also ignored, hence when copying the submodule to `/tmp`, `release-plz` skips these files and then the dirty check fails because the submodule now is not clean. As a workaround I have now enabled `allow_dirty`.
My question is whether there is a neater solution to this. Is there a way to only allow certain subdirectories to be dirty, or can `release-plz` somehow be extended to avoid this issue, at least on submodules that one might not have control over to modify `.gitignore`. Thanks.

**Comment by marcoieni** [comment, marked as answer]
Task 2 of issue https://github.com/release-plz/release-plz/issues/721#issue-1680181256 should solve this

**Reply by marcoieni** [reply]
PR welcome 👍
